### PR TITLE
HDDS-9092. Recon Legend Text Alignment in Disk Usage

### PR DESCRIPTION
**Pull request by smitajoshi12**
## What changes were proposed in this pull request?
1) Recon legend alignment issue in CSS.
2) Closing option for Metadata Summary Information Drawer.

## What is the link to the Apache JIRA
https://issues.apache.org/jira/browse/HDDS-9092

## How was this patch tested?
Manually
Previous  Issue
image

New Changes with this PR
image

Testing with Responsive Web
image

Testing with Big Screen Size
Dimension 2500 * 1000
image

Testing with Small Screen Size
image

Closing Option Enabled for Metadata Summary Drawer
image

### CODE REVIEW

File: hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/diskUsage/diskUsage.less [hunk 1]
```diff
@@ -79,4 +79,16 @@
 
 .ant-drawer-open .ant-drawer-mask {
   opacity: 0 !important;
+}
+
+div.svg-container {
+  width: 1400px !important;
```
Comment (devmadhuu) [inline hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/diskUsage/diskUsage.less:85]: Is hardcoding this width here compatible with all displays and large displays ? Shouldn't this be relative to screen size ?

File: hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/diskUsage/diskUsage.less [hunk 2]
```diff
@@ -79,4 +79,16 @@
...
+}
+
+svg.main-svg {
+  width: 1400px !important;
```
Comment (devmadhuu) [inline hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/diskUsage/diskUsage.less:89]: Same comment as above.

File: hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/diskUsage/diskUsage.less [hunk 3]
```diff
@@ -79,4 +79,16 @@
...
+}
+
+g.legend {
+  transform: translate(650px, 100px) !important;
```
Comment (devmadhuu) [inline hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/diskUsage/diskUsage.less:93]: Same comment as above.

File: hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/diskUsage/diskUsage.less (hunk 1 above)
Comment (smitajoshi12) [reply hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/diskUsage/diskUsage.less:85]: @devmadhuu 
We have tested different screensize and Responsive web also. Attached new screenshots with testing result.

File: hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/diskUsage/diskUsage.less (hunk 2 above)
Comment (smitajoshi12) [reply hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/diskUsage/diskUsage.less:89]: @devmadhuu 
We have tested different screensize and Responsive web also. Attached new screenshots with testing result.

File: hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/diskUsage/diskUsage.less (hunk 3 above)
Comment (smitajoshi12) [reply hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/diskUsage/diskUsage.less:93]: @devmadhuu 
We have tested different screensize and Responsive web also. Attached new screenshots with testing result.

**Review by devmadhuu — APPROVED** [review]
Thanks @smitajoshi12  LGTM +1

File: hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/diskUsage/diskUsage.less (hunk 1 above)
Comment (dombizita) [reply hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/diskUsage/diskUsage.less:85]: why is this hardcoded value will be good? what happens if you are checking it on smaller displays?

**Review by dombizita — COMMENTED** [review]
thanks for working on this @smitajoshi12! in the description in the `Testing with Responsive Web` screenshot on the right side the white background gets bigger from the point the figure starts. I added an arrow pointing to that.

recon-screenshot

was this there before? is it happening because of this change?

File: hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/api/db.json [hunk 4]
```diff
@@ -1346,7 +1346,7 @@
     "subPathCount": 5,
     "subPaths": [
       {
-        "path": "/vol1",
+        "path": "/vol1cloudera-health-monitoring-ozone-basic-canary-volume",
```
[the remark this answers is not in the document]
Comment (smitajoshi12) [reply hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/api/db.json]: Completed Change in latest commit.

**Comment by smitajoshi12** [comment]
> thanks for working on this @smitajoshi12! in the description in the `Testing with Responsive Web` screenshot on the right side the white background gets bigger from the point the figure starts. I added an arrow pointing to that.
> 
> recon-screenshot
> 
> was this there before? is it happening because of this change?

Yes It is existing thing when we do screen minimization as ccss properties are getting overlapped on each other.

**Comment by dombizita** [comment]
> Yes It is existing thing when we do screen minimization as ccss properties are getting overlapped on each other.

could you please check what causes this issue? is it easily solvable?

**Comment by dombizita** [comment]
Could you take a look at this patch @devabhishekpal @ivandika3? thanks in advance!!

**Comment by ivandika3** [comment]
@smitajoshi12 Thank you for working on this.

I don't have a lot of experience and knowledge in CSS tuning. From my understanding, instead of adjusting the hardcoded CSS value, it's usually better to use Ant Design 24 Grids system (https://ant.design/components/grid) to setup the layout of the page. It also has responsive feature by setting up the ratio based on the dimensions (https://ant.design/components/grid#components-grid-demo-responsive). You might want to take a look.

Let me try to create PoC on what I meant.

**Comment by smitajoshi12** [comment]
> es It is existing thing when we do screen minimization as ccss properties are getting overlapped on each other.

@ivandika3 
Thanks Ivan

**Comment by smitajoshi12** [comment]
will close this PR instead of hardcoded CSS will check Antd grid systems.

**Comment by ivandika3** [comment]
Hi @smitajoshi12 I did some PoC that time, but was unable to make it work yet. I suspect that plotly and antd grid does not play well.

I will continue when I have time. If you can help to continue this, that would be great. 

Reference: https://github.com/ivandika3/ozone/tree/DISK-USAGE-UI-POC

**Pull request closed without merging**